src/segments/project.go

```diff
@@ -24,6 +50,7 @@ type Project struct {
 
 	projects []*ProjectItem
 	Version  string
+	Name     string
```

Comment:
I would create 1 type called ProjectData that contains these fields and is added using struct inheritance:

```go
type ProjectData struct {
    Name    string
    Version string
}

type Project struct {
    ...
 	projects []*ProjectItem
    ProjectData
```

You can then drop all project specific types, they don't really mean anything outside

Reply:
done